src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,69 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.InvalidValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_LOGIN_LENGTH = 6;
+    private static final int VALID_PASSWORD_LENGTH = 6;
+    private static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
     }
+
+    private void validateUser(User user) {
+        checkIfUserNull(user);
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        checkIfUserPresentInDataBase(user);
+
+    }
+
+    private void checkIfUserNull(User user) {
+        if (user == null) {
+            throw new InvalidValidationException("User cannot be null");
+        }
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidValidationException("User's login cannot be null");
+        }
+        if (user.getLogin().length() < VALID_LOGIN_LENGTH) {
+            throw new InvalidValidationException("Incorrect user`s login. Login must be at least "
+                    + VALID_LOGIN_LENGTH + " characters");
+        }
```

Comment:
Such differentiation of messages is usually not required.
Main goal is to give user clue what he should fix.
Also good practice is to add user's input into message so he can instantly get which value is wrong
```suggestion
        if (user.getLogin() == null || user.getLogin().length() < VALID_LOGIN_LENGTH) {
            String message = String.format("User's login is mandatory and should contain at least %s characters. Provided: '%s'",
                   VALID_LOGIN_LENGTH, user.getLogin)
            throw new InvalidValidationException(message);
        }
```